database/database.go

```diff
@@ -210,28 +210,27 @@ func (db *Database) Close() {
  * Private
  ******************************************************************************/
 
-func newDatabase(
-	name string, migrationPath string, mergedMigrationPath string, version uint) (db *Database, err error) {
+func newDatabase(name string, migrationPath string, mergedMigrationPath string, version uint) (*Database, error) {
 	log.WithField("name", name).Debug("Open database")
 
 	// Check and create db path
-	if _, err = os.Stat(filepath.Dir(name)); err != nil {
+	if _, err := os.Stat(filepath.Dir(name)); err != nil {
 		if !os.IsNotExist(err) {
-			return db, aoserrors.Wrap(err)
+			return nil, aoserrors.Wrap(err)
 		}
 
 		if err = os.MkdirAll(filepath.Dir(name), 0o755); err != nil {
-			return db, aoserrors.Wrap(err)
+			return nil, aoserrors.Wrap(err)
 		}
 	}
 
 	sqlite, err := sql.Open("sqlite3", fmt.Sprintf("%s?_busy_timeout=%d&_journal_mode=%s&_sync=%s",
 		name, busyTimeout, journalMode, syncMode))
 	if err != nil {
-		return db, aoserrors.Wrap(err)
+		return nil, aoserrors.Wrap(err)
```

Comment:
Change return nil in the following cases as well.

Reply:
fixed